docs/current_docs/manuals/developer/typescript/356352-runtime.mdx

```diff
@@ -0,0 +1,42 @@
+---
+slug: /manuals/developer/typescript/356352/runtime
+displayed_sidebar: "current"
+toc_max_heading_level: 2
+title: "Runtimes"
+---
+
+# Runtimes
+
+Multiple runtimes support TypeScript such as [Node.js](https://nodejs.org/en), [Deno](https://deno.com/) and [Bun](https://bun.sh/).
+
+By default, the TypeScript SDK executes functions using the Node.js runtime, but you can configure an alternative runtime in the `package.json` file.
+
+:::note
+For now, the TypeScript SDK only supports Node.js (stable) and Bun (experimental).
+:::
+
+To change the TypeScript SDK runtime, set the field `dagger.runtime` in the `package.json` file.
+
+## Node  
+
+Set the field to `node` to use the Node.js runtime. Node.js is also the default runtime used if this field is omitted.
+
+```json
+  "dagger": {
+    "runtime": "node"
+  }
+```
+
+## Bun 
+
+Set the field to `bun` to use the Bun runtime.
+
+```json
+  "dagger": {
+    "runtime": "bun"
+  }
+```
+
+:::warning
+Bun runtime support is still experimental. Unexpected results may occur in some cases.
+:::
```

Comment:
```suggestion
:::
## Automatic Detection
When a runtime is not explicitly defined within the `package.json` file, Dagger automatically identifies the appropriate runtime by examining the project's lock files inside the projects `dagger` directory.

- If Dagger finds any of the following lock files : `package-lock.json`, `yarn.lock`, or `pnpm-lock.yaml`, it automatically selects `node` as the runtime.

- In the absence of the lock files mentioned above, if a `bun.lockb` file is present, Dagger will choose `bun` as the runtime.

- If no or only unknown lock files are present `node` is choosen.

This behavior however should be seen as a sensible fallback and not as an explicit configuration.
```

We could add something like this at the end. This way potential confusions could be avoided. F.e someone uses bun as a package manager, but still wants to use the Node.js runtime. This section explains daggers behavior in this situation.
I am also open to not include this for the sake of simplicity.

Reply:
This seems like valuable information to add.